src/freenet/clients/http/UserAlertsToadlet.java

```diff
@@ -47,7 +50,16 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
 			int userAlertHashCode = request.getIntPart("disable", -1);
 			ctx.getAlertManager().dismissAlert(userAlertHashCode);
 		}
-		headers.put("Location", ".");
+		
+		
+		String redirect;
+		try {
+			redirect = request.getPartAsStringThrowing("redirectToAfterDisable", 1024);
+		} catch (SizeLimitExceededException | NoSuchElementException e) {
+			redirect = ".";
+		}
+		headers.put("Location", redirect);
```

Comment:
This right here is called an open-redirect vulnerability.
https://www.owasp.org/index.php/Unvalidated_Redirects_and_Forwards_Cheat_Sheet